src/app/auth/profiles/profiles.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { MatDialog } from '@angular/material';
+import { JsonParseErrorDialogComponent } from './json-parse-error-dialog/json-parse-error-dialog.component';
+const { ipcRenderer } = require('electron');
+
+/**
+ * Indicates all the elements that make up a Profile.
+ */
+export interface Profile {
+  profileName: string;
+  username: string;
+  password: string;
+  encodedText: string;
+}
+
+/**
+ * This component is responsible allowing the user to select a Profile
+ * in the authentication page.
+ */
+
+@Component({
+  selector: 'app-profiles',
+  templateUrl: './profiles.component.html',
+  styleUrls: ['./profiles.component.css']
+})
+export class ProfilesComponent implements OnInit {
+
+  profiles: Profile[] = undefined; // List of profiles taken from profiles.json
+  blankProfile: Profile = {profileName: '', password: '', username: '', encodedText: ''}; // A blank profile to reset values
+
+  private readonly fs = require('fs');
+
+  private readonly APPLICATION_AND_SUBDIRECTORIES: RegExp = /\/*[^\/]+\.(exe|app|AppImage)\/*.*/g;
+  private readonly PROFILES_FILE_NAME = 'profiles.json';
+  private filePath: string;
+
+  @Output() selectedProfile: EventEmitter<Profile> = new EventEmitter<Profile>();
+
+  constructor(public errorDialog: MatDialog) { }
+
+  ngOnInit() {
+
+    this.filePath = ipcRenderer.sendSync('synchronous-message', 'getDirectory')
+        .replace(this.APPLICATION_AND_SUBDIRECTORIES, '')
+        .concat('/')
+        .concat(this.PROFILES_FILE_NAME);
+
+    if (this.userProfileFileExists(this.filePath)) {
+      try {
+        this.profiles = JSON.parse(this.fs.readFileSync(this.filePath))['profiles'];
+      } catch (e) {
+        console.log(e);
+      }
+
+      setTimeout(() => {
+        if (!this.isValid(this.profiles)) {
+          this.openErrorDialog();
+          this.profiles = undefined;
+        }
+      });
+    } else {
+      console.log(this.filePath + ' does not exist.');
+    }
```

Comment:
Would this be okay?

```typescript

if (!this.userProfileExists(this.filePath)) {
  console.log(this.filePath + ' does not exist.');
  return;
}

try {
  this.profiles = JSON.parse(this.fs.readFileSync(this.filePath))['profiles'];
} catch (e) {
  console.log(e);
}

setTimeout(() => {
  if (!this.isValid(this.profiles)) {
    this.openErrorDialog();
    this.profiles = undefined;
  }
});
```

Reply:
Yes.
On a related note, we should add a proper logging mechanism so that users can send us the log file if it can help in troubleshooting a problem. But that can be a separate PR.

Reply:
Sure 👍